### PR TITLE
Firestore build and testing improvements and documentation

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: firestore/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+requirements.
+
+For CPP SDK to compile, you generally need your `pyenv which python` to point to
+a Python3 installation.
+
+# Architecture
+
+It is easier to work this Firestore CPP SDK by keeping a high level archetecture in mind:
+
+![architecture.png](architecture.png)
```
Comment (tom-andersen) [inline firestore/CONTRIBUTING.md:24]: I am not sure where this diagram came from, but instinctively the arrows seem to point in the wrong direction. I looked up UML diagrams, and how IntelliJ does a Dependency Diagrams (also based on UML I think), and they all seem to point towards the root of dependency.

The first question that popped into my mind when I saw this diagram was: Why does Core SDK have knowledge of Swift? Shouldn't Swift just be an agnostic wrapper around Core SDK? Then I realized that my understanding of arrows (which is consistent with UML) is completely backwards to what the diagram uses.

File: firestore/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+# Clones the origin/master branch of `firebase-ios-sdk` to get the Core SDK.
+cmake -DFIRESTORE_DEP_SOURCE=origin/master ..
+
+# Clones commit '555555' of `firebase-ios-sdk`.
+cmake -DFIRESTORE_DEP_SOURCE=555555 ..
+```
+
+## IDE Integration
+
+Open up the repo root directory from `CLion` should load all symbols for the SDK itsel should
```
Comment (tom-andersen) [inline firestore/CONTRIBUTING.md:81]: There are some repeated words:

should load all symbols for the SDK itsel

File: firestore/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+cmake -DFIRESTORE_DEP_SOURCE=origin/master ..
+
+# Clones commit '555555' of `firebase-ios-sdk`.
+cmake -DFIRESTORE_DEP_SOURCE=555555 ..
+```
+
+## IDE Integration
+
+Open up the repo root directory from `CLion` should load all symbols for the SDK itsel should 
+load all symbols for the SDK itself. Once loaded, you can right load on
```
Comment (tom-andersen) [inline firestore/CONTRIBUTING.md:82]: I think `right load` should be `right click` ? What option from drop down menu should I choose?

File: firestore/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+## IDE Integration
+
+Open up the repo root directory from `CLion` should load all symbols for the SDK itsel should 
+load all symbols for the SDK itself. Once loaded, you can right load on 
+`firestore/integration_test_internal/CMakeLists.txt` and `load project` to load the tests into
+the IDE.
+
+# Android building and testing
+
+Once Android NDK is installed and added to `local.properties`, and `google-services.json`,
```
Comment (tom-andersen) [inline firestore/CONTRIBUTING.md:88]: Where can I find more instruction on installing Android NDK? If I recall, there is a specific version that must be used.

**Review by tom-andersen — COMMENTED** [review]
This looks super helpful.

I noticed a few problems with wording and additions I would like to see.

File: firestore/CONTRIBUTING.md (hunk 2 above)
Comment (wu-hui) [reply firestore/CONTRIBUTING.md:81]: Fixed.

File: firestore/CONTRIBUTING.md (hunk 3 above)
Comment (wu-hui) [reply firestore/CONTRIBUTING.md:82]: Fixed.

File: firestore/CONTRIBUTING.md (hunk 4 above)
Comment (wu-hui) [reply firestore/CONTRIBUTING.md:88]: local.properties should be the same as you use for Android. Or is there anything in particular?

File: firestore/CONTRIBUTING.md (hunk 1 above)
Comment (wu-hui) [reply firestore/CONTRIBUTING.md:24]: Addressed. I am never a fan of UML, but I guess there is value to follow known conventions.

**Review by wu-hui — COMMENTED** [review]
@tom-andersen Please take another look.

**Review by tom-andersen — DISMISSED** [review]

**Review by tom-andersen — APPROVED** [review]

**Pull request merged**